modules/programs/neovim.nix

```diff
@@ -174,8 +159,14 @@ in {
         '';
       };
 
+      # Currently, upstream Neovim is pinned on Lua 5.1 for LuaJIT support.
+      # This will need to be updated if Neovim ever migrates to a newer
+      # version of Lua.
       extraLuaPackages = mkOption {
-        type = with types; either extraLua51PackageType (listOf package);
+        type = with types;
+          let finalType = listOf package;
```

Comment:
There is a reason we prefer functions, it allows to work with different sets. Actually we should use the same lua as neovim-unwrapped. It should be gettable via `programs.neovim.finalPackage.neovim.lua` so one could do 
`programs.neovim.finalPackage.neovim.lua.withPackages(ls: extraLuaPackages)`

Reply:
Yeah, agreed. I made this change now.